lib/delayed/psych_ext.rb

```diff
@@ -74,7 +74,11 @@ def visit_Psych_Nodes_Mapping(object) # rubocop:disable CyclomaticComplexity, Me
             raise Delayed::DeserializationError, "DataMapper::ObjectNotFoundError, class: #{klass} (#{error.message})"
           end
         else
-          super
+          begin
+            super
+          rescue Exception => error
```

Comment:
Unfortunately Ruby raises an Exception, not an Error, when attempting to
unmarshal a Delegator with no __setobj__ :(

On Friday, April 15, 2016, Jack Danger Canty <notifications@github.com>
wrote:

> In lib/delayed/psych_ext.rb
> <https://github.com/collectiveidea/delayed_job/pull/907#discussion_r59939272>
> :
>
> > @@ -74,7 +74,11 @@ def visit_Psych_Nodes_Mapping(object) # rubocop:disable CyclomaticComplexity, Me
> >              raise Delayed::DeserializationError, "DataMapper::ObjectNotFoundError, class: #{klass} (#{error.message})"
> >            end
> >          else
> > -          super
> > +          begin
> > +            super
> > +          rescue Exception => error
>
> I recommend rescue => error because Exception is a little too greedy and
> catches, for example, CTRL-C-ing the process. Leaving out the exception
> class implies rescue StandardError => error.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/collectiveidea/delayed_job/pull/907/files/5e922eacee9dda00c4316badfce6d9edafcfe208#r59939272>
>